File: ecommerce-backend/Models/Order.cs
using System;
using System.Collections.Generic;

namespace ecommerce_backend.Models;

public partial class Order
{
    public int OrderId { get; set; }

    public int CustomerId { get; set; }

    public DateTime OrderDate { get; set; }

    public decimal TotalAmount { get; set; }

    public string? Status { get; set; }

    public string Address { get; set; } = null!;

    public string? Phone { get; set; }

    public string? PaymentMethod { get; set; }

    public string? ShippingService { get; set; }

    public string? Notes { get; set; }

    public virtual Customer Customer { get; set; } = null!;

    public virtual ICollection<OrderDetail> OrderDetails { get; set; } = new List<OrderDetail>();
}

File: ecommerce_frontend/src/Models/Order.ts
import { VariantGet } from "./Variant";

export type CheckoutPost = {
      address: string,
      phone: string,
      paymentMethod: string,
      shippingService: string,
      notes: string
}

export interface OrderGet {
    orderId: number;
    customerName: string;
    orderDate: string;
    totalAmount: number;
    status: string | null;
    address: string;    
    phone: string | null;
    paymentMethod: string | null;
    shippingService: string | null;
    notes: string | null;
    orderDetails: OrderDetailGet[];
}

export interface OrderDetailGet {
    orderDetailId: number;
    orderId: number;
    variant: VariantGet;
    quantity: number;
    price: number;
}